### PR TITLE
Add uploadDataToGoogleSheets API

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Add uploadDataToGoogleSheets API

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/google_sheets.R [hunk 1]
```diff
@@ -27,6 +27,31 @@ updateGoogleSheet <- function(filepath, id, overwrite = FALSE){
...
 
+#' API to upload data frame to Google Sheets.
+#'
+#' @export
+#' @param df - data frame
+#' @param type - either "newSpreadSheet", "overrideSpreadSheet", "newWorkSheet", and "appendToWorkSheet"
+#' @param spreadSheetName - name of the spread sheet (when creating a new spread sheet)
+#' @param spreadSheetId - sheet id (when updating an existing spread sheet)
+#' @param workSheet - name of the worksheet
+#'
```
Comment (kei51e) [inline R/google_sheets.R:38]: Add prerequisites for using this function. I think oauth needs to be set in advance.

File: R/google_sheets.R [hunk 2]
```diff
@@ -27,6 +27,31 @@ updateGoogleSheet <- function(filepath, id, overwrite = FALSE){
...
+uploadDataToGoogleSheets <- function(df, type = "newSpreadSheet", spreadSheetName = "", workSheet = "") {
+  if(!requireNamespace("googlesheets4")){stop("package googlesheets4 must be installed.")}
+  token <- getGoogleTokenForSheet("")
```
[the remark this answers is not in the document]
Comment (hidekoji) [reply R/google_sheets.R:41]: It's not used but there for historical reasons.

File: R/google_sheets.R (hunk 1 above)
Comment (hidekoji) [reply R/google_sheets.R:41]: Added comment for pre-requisite.

File: R/google_sheets.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (hidekoji) [reply R/google_sheets.R:44]: Added comment.

**Pull request merged**